tchannel/tornado/dispatch.py

```diff
@@ -70,52 +71,55 @@ def __init__(self, _handler_returns_response=False):
         self.register(self.FALLBACK, self.not_found)
         self._handler_returns_response = _handler_returns_response
 
-    _HANDLER_NAMES = {
-        Types.CALL_REQ: 'pre_call',
-        Types.CALL_REQ_CONTINUE: 'pre_call'
-    }
-
     def handle(self, message, connection):
         # TODO assert that the handshake was already completed
         assert message, "message must not be None"
 
-        if message.message_type not in self._HANDLER_NAMES:
-            # TODO handle this more gracefully
-            raise NotImplementedError("Unexpected message: %s" % str(message))
+        if message.message_type not in self._HANDLERS:
+            raise FatalProtocolError("Unexpected message: %s" % str(message))
```

Comment:
Are you sure we want to kill the server on unexpected messages? It's probably better to log and ignore the message?

(Yeah, we've been using the NotImplementedError version so far but now that we're changing this, might as well fix it.)